mangaki/mangaki/wrappers/anilist.py

```diff
@@ -247,8 +289,20 @@ def _request(self,
             query_params = {}
 
         r = self._session.get(urljoin(self.BASE_URL, datapage.format(**params)), params=query_params)
+        if r.status_code == 404:
+            return None
         r.raise_for_status()
-        return r.json()
+
+        data = r.json()
+        # Check if data is a list of JSON objects
+        if isinstance(data, list):
+            return data
+        # Check if data is a single JSON object
+        else:
+            # Check that the JSON return is not an error (eg. No Results)
+            if data.get('error'):
+                return None
```

Comment:
I changed the way I handle these errors. Here are some examples of errors or strings returned by AniList so far :
- No results to a **search query** : `{"error":{"status":200,"messages":["No Results."]}}`
- No results to a **query by anime or manga ID** : `{"error":{"status":404,"messages":["No query results for model [App\\AniList\\v1\\Series\\Series] <ID of work>"]}}`
- No results to a **query for a user's list** : `{"error":{"status":404,"messages":["No query results for model [App\\AniList\\v1\\User\\User] <username>"]}}`
- No results for **seasonal animes** (in 2100, for instance) : `[]`

And here are more common errors not related to API routes :
- If the **API route doesn't exist** : `{"error":{"status":404,"messages":["API route not found."]}}`
- If **no access token** is provided : `{"status":401,"error":"unauthorized","error_message":"Access token is missing"}`
- If the **token expired** : `{"error":"access_denied","error_description":"The resource owner or authorization server denied the request."}`